lib/openapi_contracts/validators/headers.rb

```diff
@@ -9,6 +9,13 @@ def spec
     def validate
       spec.headers.each do |header|
         value = response.headers[header.name]
+
+        # Rack 3.0.0 returns an Array for multi-value headers. OpenAPI doesn't
+        # support multi-value headers, so we join them into a single string.
+        #
+        # @see https://github.com/rack/rack/issues/1598
+        value = value.join('\n') if value.is_a?(Array)
```

Comment:
Sorry for the long delay. But I think you need to join by `"\n"` and not `'\n'`. Single quotes would translate to actual `\n` characters instead of new line.

Reply:
Good catch. Should be fixed now.